test/k8sT/Services.go

```diff
@@ -133,15 +133,47 @@ var _ = Describe("K8sServicesTest", func() {
 		}
 	}
 
-	testCurlFromPods := func(clientPodLabel, url string) {
+	// Return a command string for bash test loop.
+	testCommand := func(cmd string, count, fails int) string {
+		// Repeat 'cmd' 'count' times, while recording return codes of failed invocations.
+		// Successful cmd exit values are also echoed for debugging this script itself.
+		// Prints "failed:" followed by colon separated list of command ordinals and exit codes.
+		// Returns success (0) if no more than 'fails' rounds fail, otherwise returns 42.
+		return fmt.Sprintf(`/bin/bash -c 'fails=""; for i in $(seq 1 %d); do if %s; then echo "Test round $i exit code: $?"; else fails=$fails:$i=$?; fi; done; if [ -n "$fails" ]; then echo "failed: $fails"; fi; cnt="${fails//[^:]}"; if [ ${#cnt} -gt %d ]; then exit 42; fi'`, count, cmd, fails)
```

Comment:
nit: Would be nice to have this defined as multiline string which has newlines replaced by semicolons for returned value. This would improve readability of this bash snippet.

Reply:
Why is it a bash script?

Reply:
@errordeveloper regex based substitution fails on Linux `/bin/sh' errors out:
```
$ /bin/sh -c 'cnt="${PATH/a/b}"'
/bin/sh: 1: Bad substitution
```
Works on bash.

Reply:
I am asking why it is a bash script as opposed to Go code?

Reply:
@nebril Done!

Reply:
I still don't quite understand why this test needs to shell out to bash to do a loop, would you mind explaining?

Reply:
@pchaigno thanks! I see the premise of that now, I think the usage of curl something to review in general. I think it would be nice to find a way of verifying networking functionality in a more high-level way without having to poke everywhere with curl or other commands, and code each case so tediously. It would be very cool to see test being driven by a declatrative communication model of some sort.

Reply:
@errordeveloper I'm tempted to answer "PRs welcomed" :stuck_out_tongue_winking_eye: